### PR TITLE
Update README

### LINKED ISSUE #21: Document visualization mechanism
When I was reviewing #17, I realized there would a need to figure out how things should be visualized precisely. Update README to tell how the different parts of a Java project, such as a method, are visualized in Surabaya.

### PR DESCRIPTION
**Pull request by algobot76**
Close #21

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,76 +12,143 @@ Follow these steps to set up the project:
...
+  - `Interface` = ???
+  - `Abstract` = ???
+  - `public` = ???
+  - `private` = ???
+  - `protected` = ???
+  - `import` = ???
+- [Field](#field-class) = ???
+- [Method](#method-class) = ???
+- [Constructor](#constructor-class) = ???
+- [Parameter](#parameter-class) = ???
```
Comment (algobot76) [inline README.md:154]: We have to think carefully about how to visualize differnet parts of Java code.

Comment (MayKikuchi) [reply README.md:154]: Currently I'm using this for the island icons. Definitely not set in stone or anything... is there something that isn't covered though, or something that would work better as an icon? Extra information like parameters and names are in the tooltip bubble: 

legendPNG

I'm having a bit of a hard time figuring out what the above syntax means though. For example,  what does '#parameter-class' in [Parameter](#parameter-class) = ??? mean, and what does the ??? stand for?

Comment (algobot76) [reply README.md:154]: `???` means the visualization of the entity is not finalized yet. `#parameter-class` is a link to the title `Parameter Class`.

Comment (algobot76) [reply README.md:154]: Java has other primitive types like `long`, `double`, etc. Do we want to visualize them as ponds or something else?

Comment (MayKikuchi) [reply README.md:154]: Sorry, I was thinking we could replace 'Object' (the rock)  with 'other'

and yeah, maybe we could depict the 8 primitive data types as an icon: https://www.w3schools.com/java/java_data_types.asp

Comment (MayKikuchi) [reply README.md:154]: actually, do we need to have an icon for 'void'? What is it used for other than in method declarations (which would be shown through the tooltip text)?

Comment (MayKikuchi) [reply README.md:154]: How about a unique icon for each of:
boolean type
numeric type
string
collection
map
other

based on this: https://www.geeksforgeeks.org/data-types-in-java/ and this: https://www.geeksforgeeks.org/collections-in-java-2/

**Comment by john-zou** [comment]
Updated the README to reflect current output of back end

**Pull request merged**